### ISSUE TITLE
Custom datatype fields types are not validated in "Firebase Schema Validation"

### ISSUE THREAD

**Issue by lcoppa** [FlutterFlow/flutterflow-issues#2945]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

I have a recipe schema, with the energy field which is a list of a custom data type. That data type contains a field of type String, BUT in the db, that fiels is actually an int. The db was filled in programmatically on the side.
Es:

recipe: {
  energy: {
    amount: <string>,           <----- this is an int in the db
    units: <string> 
  },
...
}
 
The validation does not catch the type mismatch, and all the documents containing that amunt as int are silently ignored by the firestore collection query, with NO WARNING WHATSOEVER. 
Result: those recieps are not listed in the query result.
I wasted a day getting to the bottom onof it.

### Expected Behavior

That the Validation catched the type mismatch and complains.

### Steps to Reproduce

just create the document schema and custom data type as described above.

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITFTjMmB0It2ocsA+KrPY+1Fhio4CUMdXZIRmOtvewgjF774OrA2P/SkQFdtZ9CzaHNUEVeWikY10MGPud7DLPUrBySeR70++LlxFxPPbH6iMbaYEc+sRnNSL+NUIBWcy6en3RFANu5eSi0C71yQNuiREj6Cf9qOYwx5e6fDbOY=

### Context

Just listing a bunch of documents with a simple firestore collection query 

### Visual documentation

not needed

### Additional Info

not needed

### Environment

```markdown
- FlutterFlow version: 4.1.53+ / flutter 3.19.1
- Platform: web
- Browser name and version: chrome 124.0.6367.209 (Official Build) (arm64)
- Operating system and version affected: macos 14.5
```

### General

- [ ] I was collaborating with other users in the same project prior to data loss
- [ ] I navigated between different project views leading up to data loss
- [ ] The UI was running very slowly prior to data loss
- [ ] A crash or other bug occurred prior to data loss
- [ ] Data was not saved in a snapshot
- [ ] I was using flutterflow from multiple devices
- [ ] I was using flutterflow from multiple platforms (e.g. both web and MacOS Desktop app)

### Relative to the time the changes were made, data was lost within

- [ ] 1 minute
- [ ] 1 hour
- [ ] 1 day
- [ ] 1 week

### When following my steps to reproduce, data loss happens

- [ ] ~100% of the time
- [ ] ~50% of the time
- [ ] ~10% of the time
- [ ] ~1% of the time
- [ ] Very rarely or only encountered once

**Comment by ignalauret** [comment]
Hey @Icoppa thanks for your report. I just checked you project and I can see that the amount variable inside Energy_Detail DataType is now an Integer, I hope this solved the issue. <br>About the mismatch not throwing an error, I will suggest our technical team to implement some kind of warning to be thrown on the console when parsing the DataType. <br>Thanks for the suggestion, have a great day!

**Comment by lcoppa** [comment]
Hi Ignacio, indeed now I have solved the issue. Thanks for the message!
Luca    Il martedì 21 maggio 2024 alle ore 03:57:21 CEST, Ignacio Lauret ***@***.***> ha scritto:  
 
 


Hey @icoppa thanks for your report. I just checked you project and I can see that the amount variable inside Energy_Detail DataType is now an Integer, I hope this solved the issue. 
About the mismatch not throwing an error, I will suggest our technical team to implement some kind of warning to be thrown on the console when parsing the DataType. 
Thanks for the suggestion, have a great day!
